### PR TITLE
Move tutor

### PR DESCRIPTION
**Pull request by Phlosioneer**
Fully named and documented every function and structure. Added state machine documentation. Some minor improvements to other global symbols.

### CODE REVIEW

**Comment by DizzyEggg** [comment]
This is move reminder, not move tutor.

**Comment by Phlosioneer** [comment]
Major update to this PR.

I was asked to investigate the move relearner functions in one of the assembly files, because it seemed out-of-place. Turns out that the first half of that assembly file is full of code for specialized / custom menus (e.g. pokeblock, pokenav, pokemon conditions, move relearner, and level-up-stats). I disassembled a lot of it to get to the move relearner functions. I had a lot of difficulty with the functions in that file.

**Comment by DizzyEggg** [comment]
Please keep LIST_B_PRESSED label. LIST_CANCEL is way too ambiguous.

File: sym_ewram.txt [hunk 1]
```diff
@@ -237,11 +237,7 @@ gUnknown_0203CF40: @ 203CF40
 gUnknown_0203CF44: @ 203CF44
 	.space 0x4
 
-gUnknown_0203CF48: @ 203CF48
-	.space 0x4
-
-gUnknown_0203CF4C: @ 203CF4C
-	.space 0x4
+    .include "src/menu_specialized.o"
```
Comment (DizzyEggg) [inline sym_ewram.txt]: tabs

**Comment by Phlosioneer** [comment]
LIST_B_PRESSED is inaccurate though. That number is used both when B is pressed and when whatever "Cancel" equivalent on the list is chosen (e.g. "Close case").

**Comment by huderlem** [comment]
This has conflicts now, but I'm fine with the contents of the PR.

**Pull request merged**